source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -694,92 +699,124 @@ void Graph::setRenderMaterial(UiNodePtr node)
     }
     else
     {
-        // continue downstream using output connections until a material node is found
-        std::vector<UiNodePtr> outNodes = node->getOutputConnections();
-        if (outNodes.size() > 0)
+        // Traverse downstream looking for the first renderable item found.
```

Comment:
New logic to perform traversal using Mtlx traversal. Other stopping criteria could be added or some other subset nodes could be added say for things like swatch rendering. 

If this becomes useful for general use a utility method can be added later one.